src/remote/remote_test.go

```diff
@@ -0,0 +1,271 @@
+package remote
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"testing"
+
+	pb "github.com/bazelbuild/remote-apis/build/bazel/remote/execution/v2"
+	"github.com/bazelbuild/remote-apis/build/bazel/semver"
+	"github.com/stretchr/testify/assert"
+	bs "google.golang.org/genproto/googleapis/bytestream"
+	rpcstatus "google.golang.org/genproto/googleapis/rpc/status"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/thought-machine/please/src/core"
+)
+
+func TestInit(t *testing.T) {
+	c := newClient()
+	assert.NoError(t, c.CheckInitialised())
+}
+
+func TestBadAPIVersion(t *testing.T) {
```

Comment:
It's not clear how setting the major versions to `1` here creates the expected output

Reply:
I'll add a comment.